src/app/test/business/dad_jokes_service_test.dart

```diff
@@ -1,49 +1,51 @@
 import 'dart:convert';
 
 import 'package:app/access/dad_jokes/dad_jokes_repository.dart';
+import 'package:app/access/dad_jokes/dad_jokes_repository_decorator.dart';
 import 'package:app/access/dad_jokes/data/dad_joke_content_data.dart';
 import 'package:app/access/dad_jokes/favorite_dad_jokes_repository.dart';
+import 'package:app/access/dad_jokes/favorite_dad_jokes_repository_decorator.dart';
+import 'package:app/access/dad_jokes/mocks/dad_jokes_list_mock.dart';
+import 'package:app/access/mocking/mocking_repository.dart';
 import 'package:app/business/dad_jokes/dad_joke.dart';
 import 'package:app/business/dad_jokes/dad_jokes_service.dart';
+import 'package:dio/dio.dart';
 import 'package:flutter_test/flutter_test.dart';
 import 'package:logger/logger.dart';
-import 'package:mockito/annotations.dart';
-import 'package:mockito/mockito.dart';
 import 'package:shared_preferences/shared_preferences.dart';
-import '../mocks/dad_jokes_data_mock.dart';
-import '../mocks/dad_jokes_list_mock.dart';
-import 'dad_jokes_service_test.mocks.dart';
-
-@GenerateNiceMocks(
-  [
-    MockSpec<DadJokesRepository>(),
-  ],
-)
+
+
 void main() {
-  late DadJokesRepository dadJokesRepository;
-  late FavoriteDadJokesRepository favoriteDadJokesRepository;
+  late DadJokesRepositoryDecorator dadJokesRepository;
+  late FavoriteDadJokesRepositoryDecorator favoriteDadJokesRepository;
+  late MockingRepository mockingRepository;
   late DadJokesService SUT;
 
   setUp(() {
     TestWidgetsFlutterBinding.ensureInitialized();
     SharedPreferences.setMockInitialValues({
-      "favoriteDadJokes": getFavoriteDadJokesList()
+      "favoriteDadJokes": getMockedFavoriteDadJokesList()
           .map((dadJoke) => jsonEncode(dadJoke))
           .toList()
     });
 
-    provideDummy(mockedDadJokeResponse);
-    dadJokesRepository = MockDadJokesRepository();
-    when(dadJokesRepository.getDadJokes()).thenAnswer(
-      (_) async => mockedDadJokeResponse,
+    mockingRepository = MockingRepository();
+    dadJokesRepository = DadJokesRepositoryDecorator(
```

Comment:
Since this is a test, I guess you don't really need the decorator (and especially the repository with a Dio instance) and should instead rely only on the mock implementation?

Reply:
I changed it back to mockito instead of the decorator for simplicity